extensions/hadoop/src/main/java/com/hazelcast/jet/hadoop/impl/ReadHadoopNewApiP.java

```diff
@@ -225,6 +233,30 @@ private Supplier(
                             }
                     ).collect(toList());
         }
+
+        private List<IndexedInputSplit> reAssignSplits(List<IndexedInputSplit> assignedSplits) {
+            // If the local file system is marked as shared, no re-assign
+            if (configuration.getBoolean(HadoopSources.SHARED_LOCAL_FS, false)) {
+                return assignedSplits;
+            }
+            try {
+                // If any of the input paths do not belong to LocalFileSystem, no re-assign
+                Job job = Job.getInstance(configuration);
```

Comment:
The shared FS flag is set to false, maybe we should always reassign and forbid non-local FS & shared=false combination.

Also we can avoid assigning the splits in meta supplier if we assign them here.

Reply:
I thought `SHARED_LOCAL_FS` property as a way to pass `sharedFileSystem` parameter from file source builder to hadoop implementation. it is false by default on file source builder. if it is explicitly set to true we don't do the re-assign, if it is not set then we check the filesystem and if all the inputs are local we re-assign. 

I don't have a strong opinion here, let's discuss it